nilearn/maskers/base_masker.py

```diff
@@ -422,13 +422,23 @@ def inverse_transform(self, X):
         """
         check_is_fitted(self)
 
+        self._check_signal_shape(X)
+
         img = self._cache(unmask)(X, self.mask_img_)
         # Be robust again memmapping that will create read-only arrays in
         # internal structures of the header: remove the memmaped array
         with contextlib.suppress(Exception):
             img._header._structarr = np.array(img._header._structarr).copy()
         return img
 
+    def _check_signal_shape(self, signals: np.ndarray):
+        if signals.shape[-1] != self.n_elements_:
+            raise ValueError(
+                "Input to 'inverse_transform' has wrong shape.\n"
+                f"Last dimension should be {self.n_elements_}.\n"
+                f"Got {signals.shape[-1]}."
+            )
+
```

Comment:
If I'm not mistaken, this function is exactly the same as the one starting at line 627. Isn't it possible to define it once in a base class, or else where and call that function?

Reply:
yeah that's because we got a base class for surface masker and one for volume masker

I have been thinking of extracting commonalities in a base class but so far the amount of duplication is bearable in my opinion but definitely something to keep a eye on.

Reply:
May be putting the function in a common utility class and call that function from `_check_signal_shape` to avoid writing the same code twice?

Reply:
yeah I will do that in the follow up PR: I need to refactor how some of the lut code is handled across label maskers